core/ledger/kvledger/txmgmt/statedb/statecouchdb/cache.go

```diff
@@ -64,9 +65,12 @@ func (c *cache) getState(chainID, namespace, key string) (*CacheValue, error) {
 	}
 
 	cacheKey := constructCacheKey(chainID, namespace, key)
-	valBytes := cache.Get(nil, cacheKey)
+	valBytes := cache.GetBig(nil, cacheKey)
```

Comment:
To clear one doubt, I had a quick look into the code of this library and unfortunately, it turned out to be correct. Blindly look for a key using `GetBig` API can cause unintended consequences. I was able to construct one example that leads to a panic.

```
func TestDoesItWork(t *testing.T) {
	c := fastcache.New(64 * 1024 * 1024)
	c.Set([]byte("key"), []byte("sixteen-byte-val"))
	fmt.Printf("[%s]\n", c.GetBig(nil, []byte("key")))
}
```
It appears that the library assumes that the application remembers what keys were stored using `Set` API V/s `SetBig` API.

In order to fix the bug, my suggestion would be... to check the value size, if greater than size limit then delete the existing key from cache if present